src/main/java/io/aiven/kafka/connect/opensearch/DocumentIDStrategy.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2019 Aiven Oy
+ * Copyright 2016 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.opensearch;
+
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import org.opensearch.action.DocWriteRequest;
+import org.opensearch.action.delete.DeleteRequest;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.index.VersionType;
+
+public enum DocumentIDStrategy {
+
+    NONE("none", "No Doc ID is added",
+         record -> null,
+         (request, record) -> Objects.isNull(record.value()) ? null : request
+    ),
+    
+    RECORD_KEY("record.key", "Generated from the record's key",
+        record -> RecordConverter.convertKey(record.keySchema(), record.key()),
+        (request, record) ->  {
+            request.versionType(VersionType.EXTERNAL);
+            request.version(record.kafkaOffset());
+            return request;
+        }
+    ),
+
+    TOPIC_PARTITION_OFFSET("topic.partition.offset", "Generated as record's ``topic+partition+offset``",
+        record ->  record.topic() + "+" + record.kafkaPartition() + "+" + record.kafkaOffset(),
+        (request, record) -> request
+    );
+
+            
+    private final String name;
+
+    private final String description;
+
+    private final Function<SinkRecord, String> docIdGenerator;
+
+    private final BiFunction<DocWriteRequest<?>, SinkRecord, DocWriteRequest<?>> updateRequest;
+
+    private DocumentIDStrategy(final String name, final String description, 
+                               final Function<SinkRecord, String> docIdGenerator,
+                               final BiFunction<DocWriteRequest<?>, SinkRecord, DocWriteRequest<?>> updateRequest) {
```

Comment:
Super minor, since this is function, `updateRequest` -> `requestUpdater` would be easier to read